public/index.html

```diff
@@ -8,39 +8,196 @@
+    	}
+	}
+
+	function drawRect(x,y,w,h,color) {
+		app.context.fillStyle = color;
+		app.context.fillRect(x,y,w,h);
+	}
+
+	function drawCircle(x,y,r,color) {
+		app.context.fillStyle = color;
+		
+		app.context.beginPath();
+		app.context.arc(x,y,r,0,Math.PI*2,false);
+		app.context.closePath();
+		app.context.fill();
+	}
+
+	function drawText(text,x,y,color) {
+		app.context.fillStyle = color;
+		app.context.font = "45px fantasy";
+		app.context.fillText(text,x,y); 
+	}
+
+		app.render = function(){
+			drawRect(0,0,app.width,app.height,"black");
+
+			drawNet();
+
+			drawText(player1.score,app.width/4,app.height/5,"yellow");
+			drawText(player2.score,3*app.width/4,app.height/5,"green");
+
+			//draw player paddles
+			drawRect(player1.x,player1.y,player1.width,player1.height,player1.color);
+			drawRect(player2.x,player2.y,player2.width,player2.height,player2.color);
+
+			drawCircle(ball.x,ball.y,ball.radius,ball.color);
+		}
+
+
+		//control the user paddle
+		document.addEventListener("keydown",buttonPressed);
+
+		function buttonPressed(e) {
+			//w
+			if(e.keycode == 87 || e.which == 87) {
+				player1.y -= paddleVelocity;
+			}
+			//s
+			if (e.keycode == 83 || e.which == 83) {
+				player1.y += paddleVelocity;
 			}
+			//uparrow
+			if (e.keycode == 38 || e.which == 38) {
+				player2.y -= paddleVelocity;
+			}
+			//downarrow
+			if (e.keycode == 40 || e.which == 40) {
+				player2.y += paddleVelocity;
+			}
+		}
+
+		function collisionDetection(b,p) {
+			//For ball
+			b.top = b.y - b.radius;
+			b.bottom = b.y + b.radius;
+			b.left = b.x - b.radius;
+			b.right = b.x + b.radius;
+
+			//For Player
+			p.top = p.y;
+			p.bottom = p.y + p.height;
+			p.left = p.x;
+			p.right = p.x + p.width;
+
+			return b.right > p.left && b.bottom > p.top && b.left < p.right && b.top < p.bottom;    //collision is true
+		}
+
+
+		function update() {
+			ball.x += ball.velocityX;
+			ball.y += ball.velocityY;
+
+			if(ball.y + ball.radius > app.height || ball.y - ball.radius < 0){
+        		ball.velocityY = -ball.velocityY;
+				wall.play();
+    		}
+
+			let player = (ball.x < app.width/2) ? player1 : player2;
+			if(collisionDetection(ball,player)){
+				//play sound
+				hit.play();
+
+				//check whether ball hits the player paddle or not
+				let collidePoint = ball.y - (player.y + player.height/2);
+				collidePoint = collidePoint/(player.height/2);
+
+				let radAngle = collidePoint * Math.PI/4;
+
+				//X direction of the ball
+				let dir = (ball.x < app.width/2) ? 1 : -1; 
+
+				ball.velocityX = dir*ball.speed * Math.cos(radAngle);
+				ball.velocityY = ball.speed * Math.sin(radAngle);
+
+				ball.speed += 0.5;
+				//increase the paddle speed
+			}
+			if (ball.x - ball.radius < 0)
+			{
+				//player 2 win
+				player2.score++;
+				p2Score.play();
+				resetBall();
+			}else if (ball.x + ball.radius > app.width)
+			{
+				//player 1 win
+				player1.score++;
+				p1Score.play();
+				resetBall();
+			} 
```

Comment:
How do you sometimes write
```
if () {
```
sometimes
```
if (){
```
sometimes
```
if ()
{
```
?

Reply:
Sometimes I add comments for my reference and remove them later on. That is you can see the variation in my code.
I usually put comment statement like this:
if () { //comment goes here for my reference
}.

Reply:
My code initially looks messy because usually, I try to create a working prototype and that is where I don't usually beautify the code I do those tasks later on after achieving the end result.